<h1 style="text-align:center;font-size:200%;;">Heart Attack</h1>
<h2 style="text-align:center;font-size:100%;;">Classification using DT</h2>
<h4 style="text-align:left;">Réaliser par  :Marouan Daghmoumi & Abderrazzak El Bourkadi & </h4>
<h4 style="text-align:left;">Encadré par  : Pr. Abdellah AZMANI</h4>

## 1. Importation des librairies

In [1]:
import seaborn as sns
import numpy as np # linear algebra
import pandas as pd # data processing, CSV file I/O (e.g. pd.read_csv)
import matplotlib.pyplot as plt
from sklearn.model_selection import train_test_split
from sklearn.preprocessing import StandardScaler 
from sklearn.svm import SVC
from sklearn.linear_model import LogisticRegression
import xgboost as xgb
from sklearn.metrics import accuracy_score
from sklearn.neighbors import KNeighborsClassifier  
from sklearn.ensemble import RandomForestRegressor
from sklearn.neighbors import KNeighborsClassifier
from sklearn.naive_bayes import BernoulliNB
from sklearn.naive_bayes import GaussianNB
from sklearn.metrics import confusion_matrix

## 2. Collecte des données

In [2]:
data=pd.read_csv('heart.csv')

In [3]:
print(data)

     age  sex  cp  trtbps  chol  fbs  restecg  thalachh  exng  oldpeak  slp  \
0     63    1   3     145   233    1        0       150     0      2.3    0   
1     37    1   2     130   250    0        1       187     0      3.5    0   
2     41    0   1     130   204    0        0       172     0      1.4    2   
3     56    1   1     120   236    0        1       178     0      0.8    2   
4     57    0   0     120   354    0        1       163     1      0.6    2   
..   ...  ...  ..     ...   ...  ...      ...       ...   ...      ...  ...   
298   57    0   0     140   241    0        1       123     1      0.2    1   
299   45    1   3     110   264    0        1       132     0      1.2    1   
300   68    1   0     144   193    1        1       141     0      3.4    1   
301   57    1   0     130   131    0        1       115     1      1.2    1   
302   57    0   1     130   236    0        0       174     0      0.0    1   

     caa  thall  output  
0      0      1       1  

## About this Dataset

- **Age**: Age of the patient

- **Sex**: Sex of the patient

- **exang**: Exercise induced angina (1 = yes; 0 = no)

- **ca**: Number of major vessels (0-3)

- **cp**: Chest Pain type

  - Value 1: Typical angina
  - Value 2: Atypical angina
  - Value 3: Non-anginal pain
  - Value 4: Asymptomatic

- **trtbps**: Resting blood pressure (in mm Hg)

- **chol**: Cholesterol in mg/dl fetched via BMI sensor

- **fbs**: Fasting blood sugar > 120 mg/dl (1 = true; 0 = false)

- **rest_ecg**: Resting electrocardiographic results

  - Value 0: Normal
  - Value 1: Having ST-T wave abnormality (T wave inversions and/or ST elevation or depression of > 0.05 mV)
  - Value 2: Showing probable or definite left ventricular hypertrophy by Estes' criteria

- **thalach**: Maximum heart rate achieved

- **target**: 

  - 0 = Less chance of heart attack
  - 1 = More chance of heart attack


## 3. exploration et visualisation des données

In [4]:
data.head(10)

age  sex  cp  trtbps  chol  fbs  restecg  thalachh  exng  oldpeak  slp  \
0   63    1   3     145   233    1        0       150     0      2.3    0   
1   37    1   2     130   250    0        1       187     0      3.5    0   
2   41    0   1     130   204    0        0       172     0      1.4    2   
3   56    1   1     120   236    0        1       178     0      0.8    2   
4   57    0   0     120   354    0        1       163     1      0.6    2   
5   57    1   0     140   192    0        1       148     0      0.4    1   
6   56    0   1     140   294    0        0       153     0      1.3    1   
7   44    1   1     120   263    0        1       173     0      0.0    2   
8   52    1   2     172   199    1        1       162     0      0.5    2   
9   57    1   2     150   168    0        1       174     0      1.6    2   

   caa  thall  output  
0    0      1       1  
1    0      2       1  
2    0      2       1  
3    0      2       1  
4    0      2       1  
5    0      1       1  
6    0      2       1  
7    0      3       1  
8    0      3       1  
9    0      2       1

In [5]:
data.tail(10)

age  sex  cp  trtbps  chol  fbs  restecg  thalachh  exng  oldpeak  slp  \
293   67    1   2     152   212    0        0       150     0      0.8    1   
294   44    1   0     120   169    0        1       144     1      2.8    0   
295   63    1   0     140   187    0        0       144     1      4.0    2   
296   63    0   0     124   197    0        1       136     1      0.0    1   
297   59    1   0     164   176    1        0        90     0      1.0    1   
298   57    0   0     140   241    0        1       123     1      0.2    1   
299   45    1   3     110   264    0        1       132     0      1.2    1   
300   68    1   0     144   193    1        1       141     0      3.4    1   
301   57    1   0     130   131    0        1       115     1      1.2    1   
302   57    0   1     130   236    0        0       174     0      0.0    1   

     caa  thall  output  
293    0      3       0  
294    0      1       0  
295    2      3       0  
296    0      2       0  
297    2      1       0  
298    0      3       0  
299    0      3       0  
300    2      3       0  
301    1      3       0  
302    1      2       0

In [6]:
data.shape

(303, 14)

##### Alors on a 13 features et 303 exemple

In [7]:
data.shape


(303, 14)

### 1. peaple more chance of heart attack

In [9]:
attack = data [ data['output']==1  ]
attack 

age  sex  cp  trtbps  chol  fbs  restecg  thalachh  exng  oldpeak  slp  \
0     63    1   3     145   233    1        0       150     0      2.3    0   
1     37    1   2     130   250    0        1       187     0      3.5    0   
2     41    0   1     130   204    0        0       172     0      1.4    2   
3     56    1   1     120   236    0        1       178     0      0.8    2   
4     57    0   0     120   354    0        1       163     1      0.6    2   
..   ...  ...  ..     ...   ...  ...      ...       ...   ...      ...  ...   
160   56    1   1     120   240    0        1       169     0      0.0    0   
161   55    0   1     132   342    0        1       166     0      1.2    2   
162   41    1   1     120   157    0        1       182     0      0.0    2   
163   38    1   2     138   175    0        1       173     0      0.0    2   
164   38    1   2     138   175    0        1       173     0      0.0    2   

     caa  thall  output  
0      0      1       1  
1      0      2       1  
2      0      2       1  
3      0      2       1  
4      0      2       1  
..   ...    ...     ...  
160    0      2       1  
161    0      2       1  
162    0      2       1  
163    4      2       1  
164    4      2       1  

[165 rows x 14 columns]

In [ ]:
print('Le nombre total des personnes est :', len(data))